### ISSUE TITLE
[Feature request] Add to new Windows 11 context menu

### ISSUE THREAD

**Issue by tfiers** [eltos/PasteIntoFile#16]
I finally updated to Windows 11, it looks really slick.

Famously there's a new right-click context menu in Explorer:
<img width=600 src="https://user-images.githubusercontent.com/3773558/208628154-127ee206-71d6-459a-9b91-ae81dfc43a2d.png">

To use PasteIntoFile (whic btw works perfectly fine for me on Win11),
you have to do an extra click to open the old context menu ("Show more options"). 

(You can also press shift-F10 in Explorer, to open the old context menu directly. That requires moving from keyboard to mouse though).

It does seem possible to customize the new context menu (see the "Open in Terminal" items in the screenshot).
Maybe it's as simple as making new registry keys, like [here](https://techviral.net/add-programs-context-menu-windows-11/).

**Comment by eltos** [comment]
I think they intensionally "cleaned up" the context menu, but there are already tools out there to bring back the "old" one 🤣  
I don't have a Win11 machine to test, but I'll accept a PR if you want to implement it. It should probably be configurable by the user.

**Comment by tfiers** [comment]
Yes, that's what I've done in the end, turning off the new context menu :p
So I'm probably not gonna implement this.

If anyone's interested in the specific tool I used for that, it's part of the (very impressive) https://github.com/valinet/ExplorerPatcher

**Comment by eltos** [comment]
This seems utterly complicated. Gathering information

Question at learn.microsoft.com: https://learn.microsoft.com/en-us/answers/questions/2225674/


From https://github.com/dnGrep/dnGrep/issues/849#issuecomment-1522310231

> The PowerToys - Power Rename has a working Windows 11 context menu using native C++ code. It looks like working implementations are based on the [PhotoStoreContextMenu](https://github.com/microsoft/AppModelSamples/tree/master/Samples/SparsePackages/PhotoStoreContextMenu) sample app.
>
> Also saving this link for future reference: [Creating a sparse signed Win32 app package](https://andrewleader.medium.com/creating-a-sparse-signed-win32-app-package-9809320cfaab).


From https://community.mp3tag.de/t/context-menu-on-windows-11/54629/7
> For someone who is looking into implementing this for their own shell extension, here is an outline of the steps involved in adding support for the Windows 11 context menu:
> 1. Implementing IExplorerCommand
> 2. Creating a sparse AppxManifest.xml with <uap10:AllowExternalContent>true</uap10:AllowExternalContent> and linking to the shell extension DLL from <com:Extension Category="windows.comServer">
> 3. Creating the sparse package via MakeAppX.exe
> 4. Signing the sparse package via signtool.exe and a valid code-signing certificate
> 5. Implementing a custom installer library which uses PackageManager from [Windows.Management.Deployment](https://docs.microsoft.com/en-us/uwp/api/windows.management.deployment?view=winrt-22000) to register the signed package
> 6. Adding means to unregister the package during deinstallation
> 
> I've wasted a lot of time on this, so if you're stuck in one of the steps please let me know.


The mention of "valid code-signing certificate" worries me, if that holds true, it's a show-stopper. See https://github.com/eltos/PasteIntoFile/discussions/67

**Comment by insane66613** [comment]
Just a quick note regarding the code-signing hurdle mentioned here (for sparse packaging and Windows 11 context menu support) — there's a great free option available specifically for open-source projects:

✅ Free Code Signing via SignPath.io
SignPath Foundation offers free, secure code signing certificates for open-source projects that meet the following criteria:

Uses an OSI-approved open-source license (e.g. MIT, GPL, etc.)

Transparent public repo with automated build process

No dual-licensing or commercial clauses

Highlights:

🔐 Private keys never leave SignPath — signing is done via a secure HSM

⚙️ Integrates well with GitHub Actions (or other CI) for fully automated signed builds

🛡️ Helps eliminate Windows SmartScreen warnings and is valid for MSIX packaging

This is the same setup used by other open-source WinForms apps like [LGTV Companion](https://github.com/JPersson77/LGTVCompanion), which had similar signing needs.

Could be a practical solution if signing continues to be a blocker for modern Windows packaging. I’d be happy to help with GitHub Actions config if needed.

**Comment by Vader77Darth** [comment]
In the meantime, this free tool https://apps.microsoft.com/detail/9PC7BZZ28G0X?hl=en-us&gl=GB&ocid=pdpshare allowed me to add basically all of the existing context menu actions into the new one.

<img width="566" height="739" alt="Image" src="https://github.com/user-attachments/assets/378d2e36-0b1b-4dec-9027-86662ae47f8a" />
